model/buildlogger.go

```diff
@@ -154,16 +154,19 @@ func (l *Log) Append(ctx context.Context, lines []LogLine) error {
 	}
 	key := fmt.Sprint(util.UnixMilli(time.Now()))
 
-	linesCombined := make([]string, len(lines))
-	for idx, line := range lines {
+	lineBuffer := bytes.NewBuffer([]byte{})
+	for _, line := range lines {
 		// unlikely scenario, but just in case priority is out of range.
 		if line.Priority > level.Emergency {
 			line.Priority = level.Emergency
 		} else if line.Priority < level.Invalid {
 			line.Priority = level.Trace
 		}
 
-		linesCombined[idx] = prependPriorityAndTimestamp(line.Priority, line.Timestamp, line.Data)
+		_, err := lineBuffer.WriteString(prependPriorityAndTimestamp(line.Priority, line.Timestamp, line.Data))
```

Comment:
should we ensure that we don't miss a newline?

Reply:
what do you mean?